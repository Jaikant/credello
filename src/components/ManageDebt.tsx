import * as React from 'react';
import Link from 'next/link';
import styled from 'styled-components';
import { device } from 'libs/device';
import { useRouter } from 'next/router';
import Button from '@material-ui/core/Button';
import { TextBlack, TextBold, TextMedium } from 'components/Typography';

const ManageDebtGrid = styled.div`
  display: grid;
  grid-gap: 8px;
  grid-template-rows: 1fr 1fr;
  grid-template-columns: 1fr 1fr;
  @media ${device.tablet} {
    grid-gap: 32px;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr;
  }
  @media ${device.laptop} {
    grid-gap: 32px;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr;
  }
`;

interface HeroButtonProps {
  readonly src: string;
}

const HeroButton = styled.div<HeroButtonProps>`
  display: flex;
  justify-content: center;
  background-color: #15db95;
  background-image: ${({ src }) =>
    `url('${require(`../assets/images/${src}`)}')`};
  background-size: 100% 100%, cover;
  background-repeat: no-repeat;
  background-position: center;
  text-transform: uppercase;
  border-radius: 5px;
  ${TextBold} {
    display: grid;
    text-align: center;
    align-content: center;
    justify-content: center;
    margin: 24px;
  }
  &:before {
    content: '';
    display: block;
    height: 0px;
    width: 0px;
    background: red;
    padding-bottom: calc(90 / 164 * 100%);
  }
`;

const UnderLine = styled(TextMedium)`
  text-decoration: underline;
`;

const LinkWrapper = styled.div`
  margin: 8px;
  @media ${device.tablet} {
    margin: 16px;
  }
  @media ${device.laptop} {
    margin: 32px;
  }
`;

const ManageDebt = () => {
  const router = useRouter();

  return (
    <div>
      <TextBold mb="10px" lh="20px" color="#02254d">
        Manage your debt better
      </TextBold>
      <ManageDebtGrid>
        <HeroButton src="payoff-debt.png">
          <TextBlack size="14px" lh="18px" color="#fff">
            Ways to pay off debt
          </TextBlack>
        </HeroButton>
        <HeroButton src="estimate-debt.png">
          <TextBlack size="14px" lh="18px" color="#fff">
            Estimate debt-free date
          </TextBlack>
        </HeroButton>
        <HeroButton
          src="consolidate.png"
          onClick={() => router.push('/debt-consolidation')}
        >
          <TextBlack size="14px" lh="18px" color="#fff">
            Ways to consolidate
          </TextBlack>
        </HeroButton>
        <HeroButton src="estimate-savings.png">
          <TextBlack size="14px" lh="18px" color="#fff">
            Estimate savings
          </TextBlack>
        </HeroButton>
      </ManageDebtGrid>
      <LinkWrapper>
        <Link href="/LearnAboutDebt" passHref>
          <UnderLine size="12px" lh="16px" mt="12px" color="#5cdb95" center>
            Learn more about debt consolidation
          </UnderLine>
        </Link>
      </LinkWrapper>
    </div>
  );
};

ManageDebt.defaultProps = {};

export default ManageDebt;
