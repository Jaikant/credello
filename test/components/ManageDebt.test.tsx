import React from 'react';
import { render } from '../testUtils';
import ManageDebt from '../../src/components/ManageDebt';
import 'jest-styled-components';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<ManageDebt />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
