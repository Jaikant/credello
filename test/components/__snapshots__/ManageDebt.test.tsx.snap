// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  .c1 {
  font-family: Roboto-Bold;
  font-weight: 700;
  font-size: 16px;
  line-height: 20px;
  margin: 0;
  margin-bottom: 10px;
  margin-top: 0;
  color: #02254d;
  text-align: initial;
}

.c4 {
  font-family: Roboto-Black;
  font-weight: 700;
  font-size: 14px;
  line-height: 18px;
  margin: 0;
  margin-bottom: 0;
  margin-top: 0;
  color: #fff;
  text-align: initial;
}

.c2 {
  display: grid;
  grid-gap: 8px;
  grid-template-rows: 1fr 1fr;
  grid-template-columns: 1fr 1fr;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #15db95;
  background-image: url('test-file-stub');
  background-size: 100% 100%,cover;
  background-repeat: no-repeat;
  background-position: center;
  text-transform: uppercase;
  border-radius: 5px;
}

.c3 .c0 {
  display: grid;
  text-align: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 24px;
}

.c3:before {
  content: '';
  display: block;
  height: 0px;
  width: 0px;
  background: red;
  padding-bottom: calc(90 / 164 * 100%);
}

.c6 {
  font-family: Roboto-Medium;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
  margin: 0;
  margin-bottom: 0;
  margin-top: 12px;
  color: #5cdb95;
  text-align: center;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  margin: 8px;
}

@media (min-width:768px) {

}

@media (min-width:1224px) {
  .c1 {
    line-height: 1.5;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 14px;
  }
}

@media (min-width:1224px) {
  .c4 {
    line-height: 1.5;
    font-size: 14px;
  }
}

@media (min-width:768px) {
  .c2 {
    grid-gap: 32px;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr;
  }
}

@media (min-width:1224px) {
  .c2 {
    grid-gap: 32px;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr;
  }
}

@media (min-width:768px) {
  .c6 {
    font-size: 12px;
  }
}

@media (min-width:1224px) {
  .c6 {
    font-size: 12px;
  }
}

@media (min-width:768px) {
  .c5 {
    margin: 16px;
  }
}

@media (min-width:1224px) {
  .c5 {
    margin: 32px;
  }
}

<div>
    <h3
      class="c0 c1"
      color="#02254d"
    >
      Manage your debt better
    </h3>
    <div
      class="c2"
    >
      <div
        class="c3"
        src="payoff-debt.png"
      >
        <h3
          class="c0 c4"
          color="#fff"
        >
          Ways to pay off debt
        </h3>
      </div>
      <div
        class="c3"
        src="estimate-debt.png"
      >
        <h3
          class="c0 c4"
          color="#fff"
        >
          Estimate debt-free date
        </h3>
      </div>
      <div
        class="c3"
        src="consolidate.png"
      >
        <h3
          class="c0 c4"
          color="#fff"
        >
          Ways to consolidate
        </h3>
      </div>
      <div
        class="c3"
        src="estimate-savings.png"
      >
        <h3
          class="c0 c4"
          color="#fff"
        >
          Estimate savings
        </h3>
      </div>
    </div>
    <div
      class="c5"
    >
      <div
        class="Typography__TextMedium-sc-12f2dd0-1 c6"
        color="#5cdb95"
        href="/LearnAboutDebt"
      >
        Learn more about debt consolidation
      </div>
    </div>
  </div>
</DocumentFragment>
`;
